File: boj/python/10610.py
# 10610번: 30
# https://www.acmicpc.net/problem/10610
# Version: Python 3.9.7


from sys import stdin

read = stdin.readline

if __name__ == "__main__":
    number_list = sorted(read().rstrip(), reverse=True)
    if sum(map(int, number_list)) % 3 != 0 or "0" not in number_list:
        print(-1)
    else:
        print("".join(number_list))

File: boj/swift/10610.swift
//
//  10610번: 30
//  https://www.acmicpc.net/problem/10610
//
//  Created by WhiteHyun on 2021/11/21.
//

let number = readLine()!
let numberArray = number.sorted().reversed().map { String($0) }
if !numberArray.contains("0") || numberArray.map({ Int($0)! }).reduce(0, +) % 3 != 0 {
    print(-1)
} else {
    print(numberArray.joined())
}
